### ISSUE TITLE
Heavy artifacting after 5

### ISSUE THREAD

**Issue by rmsa5** [proconsule/nxmp#89]
NXMP 0.8.2 title overridden, FW 17.0.0, AMS 1.6.2

Playing back files (tried with several X265 MKV files) will work for about 5min30sec without any issue, then will glitch: video will turn to black and white  and keep on playing while some weird static colour shadow of the last fully properly rendered video frame will stay visible overlaid.
Here is an example:

One another occasion the frame froze in black and white while the weird coloured overlay like colour subchannel continued to play along in sync with the sound.

After quitting video (pressing B), and launching the video file again (and resuming) playback works again correctly, for again about 5-6min, then it glitches again.

On one occasion after a series of these 5-6min play-glitch-quit-resume, it became even weirder when the right joycon stopped partially responding (R still advanced video, but B didn't respond again). Homing to Horizon menu gave a weird icon telling me to press R button

I can only guess something pretty nasty with memory usage (a leak spreading even to destabilise system drivers?) Is happening in the background.

Please let me know if I can provide further info to help pin point the technical issue.

Thanks in advance for your help.
Best

**Comment by mightysashiman** [comment]
Not sure the screenshot went through:
![PXL_20231209_090910704](https://github.com/proconsule/nxmp/assets/7439992/57c4a0d2-5fd5-416c-822c-8adc7438cccb)

**Comment by mightysashiman** [comment]
Might I add: no issue playing back the same files on VLC on other devices or even on the switch through switchroot

**Comment by proconsule** [comment]
Thanks i am aware of this problem on 0.8.2
Is fixed now on 0.8.3 (unreleased as stable)

I will release soon a new stable version, if you want to try the OpenBeta of 0.8.3 look at the NXMP thread on GBATemp forum.

Thanks.

**Comment by uncle-ben-devel** [comment]
In 0.8.3 and 0.9, the file that exibited this behavior (H.264, 10 bit), is not not playing any more.
(It says its buffering. Sometimes, NXMP freezes).
It's working without artifacting in 0.7, and it's also playing.

**Comment by Rolen28** [comment]
I can confirm this issue is still present in version 0.9

**Comment by mightysashiman** [comment]
> I can confirm this issue is still present in version 0.9

Yes I confirm too. Though, it would seem it is only happening in the opengl version, not the deko3d one (further testing to check required though)

**Comment by xfangfang** [comment]
@proconsule 
I have encountered the similar issue when developing wiliwili. I just found that setting `opengl-glfinish` to `yes` could avoid the problem (not quite sure about the reason, but I guess it might be an issue with Mesa on Switch).

https://mpv.io/manual/stable/#options-opengl-glfinish
https://mpv.io/manual/stable/#low-latency-playback

> --opengl-glfinish=yes, can reduce buffering in the graphics driver

Although according to the mpv documentation, this would greatly affect performance, my actual tests showed no observable difference compared to before.

**Comment by proconsule** [comment]
@xfangfang Thanks for this hint that can be used inide the mpv conf file. You are a great developer and i respect you a lot!

The DEKO3D version is not affected with this bug afaik.

I am very very busy in my life (work,family and so on) and i don't have too much time right now to update nxmp (i have some features never released like HW deinterlace and updated hw decoding code. But i hope next month to have a bit of free time to answer issues and maybe release an updated version of NXMP (with fixed bugs that has been reported)

**Comment by proconsule** [comment]
0.9.1 Fix all video issue so far
